src/EventSubscriber/QuerySubscriber.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Drupal\oe_list_pages\EventSubscriber;
+
+use Drupal\search_api\Event\QueryPreExecuteEvent;
+use Drupal\search_api\Event\SearchApiEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Provides an event subscriber that allows to alter list source queries.
+ */
+class QuerySubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      SearchApiEvents::QUERY_PRE_EXECUTE => 'queryAlter',
+    ];
+  }
+
+  /**
+   * Reacts to the query alter event.
+   *
+   * @param \Drupal\search_api\Event\QueryPreExecuteEvent $event
+   *   The query alter event.
+   */
+  public function queryAlter(QueryPreExecuteEvent $event) {
+    $query = $event->getQuery();
+
+    $ignored_filters = $preset_filters = [];
+
+    if ($query->getIndex()->getServerInstance()->supportsFeature('search_api_facets')) {
```

Comment:
Can we please return early if it doesn't support the feature? makes it a bit more readable.